packages/collector/test/test_util/ProcessControls.js

```diff
@@ -248,33 +256,44 @@ class ProcessControls {
             }
           })
           .join('&');
-        opts.url = opts.url.includes('?') ? `${opts.url}&${queryParams}` : `${opts.url}?${queryParams}`;
+
+        requestOpts.url = requestOpts.url.includes('?')
+          ? `${requestOpts.url}&${queryParams}`
+          : `${requestOpts.url}?${queryParams}`;
       }
-      opts.json = true;
-      opts.ca = cert;
+
+      requestOpts.json = true;
+      requestOpts.ca = cert;
+
       let response;
-      try {
-        const result = await fetch(opts.url, opts);
-        const contentType = result.headers.get('content-type');
-        if (contentType && contentType.includes('application/json')) {
-          response = await result.json();
-        } else if (contentType && (contentType.includes('text/html') || contentType.includes('text/plain'))) {
-          response = await result.text();
-        } else {
-          response = {
-            body: await result.text(),
-            status: result.status,
-            headers: result.headers
-          };
-        }
 
-        return response;
-      } catch (error) {
-        if (error.code === 'ECONNRESET' || error.type === 'request-timeout' || error.code === 'ECONNREFUSED') {
-          throw error;
+      const result = await fetch(requestOpts.url, requestOpts);
+      const contentType = result.headers.get('content-type');
+
+      if (contentType && contentType.includes('application/json')) {
+        response = await result.json();
+      } else if (contentType && (contentType.includes('text/html') || contentType.includes('text/plain'))) {
+        response = await result.text();
+      } else {
+        // CASE: Some tests do not use express and then the header is missing
+        //       Some tests use `res.send`, which is not a JSON response.
+        response = await result.text();
+      }
+
+      if (checkStatusCode) {
```

Comment:
This is a temporary workaround because I do not have the time to go through all error tests.
The correct behaviour is here to throw an error by default. If the test case throws a 500 from the app, this helper function needs to forward the error. Otherwise we could make false assumptions in the target test case. Could you please create a ticket? @aryamohanan